lib/lotus/action/cookie_jar.rb

```diff
@@ -131,6 +137,7 @@ def extract(env)
         hash   = env[COOKIE_HASH_KEY] ||= {}
         string = env[HTTP_HEADER]
 
+        return if hash.has_key? RACK_SESSION_KEY
```

Comment:
@AlfonsoUceda I tried it manually with a test application. This fix introduces another problem: it works fine for the first request, but for the second one `Set-Cookie` header is **NOT** sent at all.

/cc @pascalbetz